homeassistant/components/shelly/sensor.py

```diff
@@ -742,6 +783,8 @@ def __init__(
         """Initialize the sleeping sensor."""
         super().__init__(coordinator, block, attribute, description, entry, sensors)
 
+        self._attr_suggested_display_precision = description.suggested_display_precision
+
```

Comment:
```suggestion
```

I don't think this is needed if its set in the description

https://github.com/home-assistant/core/blob/1759f58fc110b6c3ad81d859d6e5716182ad29d2/homeassistant/components/sensor/__init__.py#L405